src/Features/Core/Portable/EventListener/ExportEventListenerAttribute.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+
+namespace Microsoft.CodeAnalysis.EventListener
+{
+    [MetadataAttribute]
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class ExportEventListenerAttribute : ExportAttribute
+    {
+        public string Service { get; }
+        public string[] WorkspaceKinds { get; }
```

Comment:
is it a good idea to expose this as a string[]?  that's mutable right, and could mean someone might change it?

Reply:
sure